### DISCUSSION TITLE
Multiple relation managers

### DISCUSSION THREAD

**Discussion by mgkimsal** [njxqlus/filament-relation-manager-component#19] [Q&A]
I don't see a way to use multiple relation managers.  The 'make()' call does not take a string/name.

Calling this multiple times only ever renders the last relation manager in the place of the first.

I don't see any examples posted of multiple relation managers in use, so it may not have been intended for that, but would really like to see that supported somehow.

Use case:  3 relation managers - wanted to move one in the middle of a form.  Leaving the other 2 in a resources' "getRelations" doesn't work - there's some JS conflict which interferes with component registration, causing "component not found for id x7ie823hew89" js errors.

Thanks!

**Comment by mgkimsal** [comment]
I think there was an issue on my side.  Was staring at this for too long.  The lack of 'name' in the relation manager wrapper threw me off some.  There was some other code in my pipeline that was filtering out some of the relation managers in my form schema.

HOWEVER...

I think there's still an issue because the relation manager components are not getting registered with livewire.

Panel::register has

```php
$this->registerLivewireComponents();
```

By default, this grabs a resource's getRelations() array, with any relation managers in there.  If they are not in there, they won't get registered in livewire, meaning none of the actions on them can be addressed as expected.  Example: hitting a resource page and having a default action trigger a mount of a table action on a relation manager.

My workaround here is to register all relation managers in getRelations() array, but render to a different template that does not include any relation manager rendering at all, then add all relation managers in the form as needed.

This is less than ideal, but I did get this working.

**Comment by njxqlus** [comment]
Not sure, what do you want to achieve, but the only usage example contains multiple relation managers
```php
public static function infolist(Infolist $infolist): Infolist
{
    return $infolist->schema([       
        Infolists\Components\Tabs::make()->tabs([
            Infolists\Components\Tabs\Tab::make('Versions')->schema([
                \Njxqlus\Filament\Components\Infolists\RelationManager::make()->manager(RelationManagers\VersionsRelationManager::class)->lazy(false)
            ]),
            Infolists\Components\Tabs\Tab::make('Stands')->schema([
                \Njxqlus\Filament\Components\Infolists\RelationManager::make()->manager(RelationManagers\StandsRelationManager::class)->lazy(false)
            ]),
            Infolists\Components\Tabs\Tab::make('Contexts')->schema([
                \Njxqlus\Filament\Components\Infolists\RelationManager::make()->manager(RelationManagers\ContextsRelationManager::class)->lazy(false)
            ]),
        ])
    ]);
}
```

**Reply by mgkimsal** [reply]
Those examples are in separate tabs.  

See my other comment regarding live wire registration.  If the relman component is just wrapped in your component, the livewire registration doesn't happen (from panel.php register() method)

**Reply by njxqlus** [reply]
Tried code:
```php
return $form
            ->schema([
                RelationManager::make()
                    ->manager(VersionsRelationManager::class)
                    ->lazy(false),
                RelationManager::make()
                    ->manager(TagsRelationManager::class)
                    ->lazy(),
```

Works fine on latest filament
